backend/ledgers.ts

```diff
@@ -12,94 +10,124 @@ interface Ledger {
+  let total = 0;
+
+  while (true) {
+    let resp = await horizon.ledgers().cursor(pagingToken).limit(200).call();
+    ledgers = resp.records;
+    total += resp.records.length;
+    if (ledgers.length == 0 || (limit && total > limit)) {
+      break;
+    }
+
+    pagingToken = ledgers[ledgers.length - 1].paging_token;
+    await updateCache(ledgers, ledgersKey, pagingTokenKey);
+  }
 }
 
-// Wait for schema sync
-postgres.sequelize.addHook("afterBulkSync", () => {
-  setInterval(updateResults, 60 * 1000);
-  updateResults();
-
-  if (process.env.UPDATE_DATA == "true") {
-    // Stream ledgers - get last paging_token/
-    postgres.LedgerStats.findOne({ order: [["sequence", "DESC"]] }).then(
-      (lastLedger: postgres.LedgerStats) => {
-        let pagingToken;
-        if (!lastLedger) {
-          pagingToken = "now";
-        } else {
-          pagingToken = lastLedger.paging_token;
-        }
-
-        var horizon = new stellarSdk.Server("https://horizon.stellar.org");
-        horizon
-          .ledgers()
-          .cursor(pagingToken)
-          .limit(200)
-          .stream({
-            // TODO - use type any until https://github.com/stellar/js-stellar-sdk/issues/731 resolved
-            onmessage: (ledger: any) => {
-              let newLedger: any = pick(ledger, [
-                "sequence",
-                "closed_at",
-                "paging_token",
-                "operation_count",
-              ]);
-              newLedger["transaction_count"] =
-                ledger.successful_transaction_count +
-                ledger.failed_transaction_count;
-              postgres.LedgerStats.create(newLedger).then(() => {
-                console.log(
-                  "Added Ledger:" + ledger.sequence + " " + ledger.closed_at,
-                );
-              });
-            },
-          });
-      },
-    );
+export async function updateCache(
+  ledgers: Array<LedgerRecord>,
+  ledgersKey: string,
+  pagingTokenKey: string,
+) {
+  if (!ledgers.length) {
+    console.log("no ledgers to update");
+    return;
   }
-});
+  const json = (await redisClient.get(ledgersKey)) || "[]";
+  let cachedLedgers: Array<Ledger> = JSON.parse(json);
+  let pagingToken = "";
+
+  ledgers.forEach((ledger: LedgerRecord) => {
+    let date = ledger.closed_at.substring(5, 10);
+    let index = findIndex(cachedLedgers, { date: date });
+    if (index == -1) {
+      cachedLedgers.push({
+        date: date,
+        transaction_count:
+          ledger.successful_transaction_count + ledger.failed_transaction_count,
+        operation_count: ledger.operation_count,
+      });
+    } else {
+      cachedLedgers.splice(index, 1, {
+        date: date,
+        transaction_count:
+          cachedLedgers[index].transaction_count +
+          ledger.successful_transaction_count +
+          ledger.failed_transaction_count,
+        operation_count:
+          cachedLedgers[index].operation_count + ledger.operation_count,
```

Comment:
maybe the naming should be changed. The cache is stored like so:
```
[
  {
    "date": "01-13",
    "transaction_count": 5245946,
    "operation_count": 10088417
  },
  {
    "date": "01-12",
    "transaction_count": 5787617,
    "operation_count": 11333624
  },
...
```
which is what `cachedLedgers` is referring to and is what's being updated. Which technically isn't a ledger, I couldn't think of a better word 🤔 . This code no longer stores every `LedgerRecord` from horizon like it did before with postgres.

I think maybe this would be clearer when LedgerRecord is not `any` anymore https://github.com/stellar/dashboard/blob/ccc4c751b63c64c604b9a8027bdb41610cedf342/backend/ledgers.ts#L13-L14

Reply:
`LedgerStat`? this was referring to something else with postgres before I removed it, but I think could be fitting here. `LedgerAggregate` might be better, but a mouth full. Thoughts on either of those?

Reply:
I like `LedgerStat` the more I think of it: https://github.com/stellar/dashboard/pull/209/commits/e6fb860861857a5c4479e4bc9b15c396c99b7e12

if you disagree lmk though